### PR TITLE
Website: Update Docusaurus to v3

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
This PR bumps Docusaurus to 3.0.0. Read more about some of the [new features here](https://docusaurus.io/blog/releases/3.0).

This PR consists of the following changes:

- Bump Docusaurus and associated dependencies to their latest versions
- [Update admonitions](https://docusaurus.io/docs/migration/v3#admonition-warning) to match older versions of Docusaurus
- Fixes [MDX build error](https://docusaurus.io/docs/migration/v3#bad-usage-of--1) in `build-a-logs-data-source-plugin.md` file due to typescript code not being wrapped in backticks.
- Define [additional languages](https://docusaurus.io/docs/migration/v3#prism-react-renderer-v20) in prism config
- Prevent errors with react/fast-refresh and tracking code by only starting tracking if production.

A future PR will migrate the configs to typescript.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md [hunk 1]
```diff
@@ -148,11 +148,11 @@ To forward requests through the Grafana proxy, you need to configure one or more
    ]
    ```
 
-  :::note
+:::note
```
Comment (sympatheticmoose) [inline docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md:151]: I think this still might be broken as it resets the numbered list

**Review by sympatheticmoose — APPROVED** [review]
LGTM only things I could find different as a result is the new italic styling on flags in `work-with-backend`
Screenshot 2023-11-03 at 14 34 57

Other comment relates to an already broken list so not a blocker

File: docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md (hunk 1 above)
Comment (sympatheticmoose) [reply docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md:151]: Sorry was looking at the wrong bit... this is fine, its further up in https://grafana-dev.com/developers/plugin-tools/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#add-secret-configuration-to-your-data-source-plugin which is broken and could be fixed

Comment (jackw) [reply docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md:151]: Ah well spotted! I'll fix that shortly. 👍

Comment (jackw) [reply docusaurus/docs/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins.md:151]: Having jumped into this I've discovered this is a bit more widespread than I first realised. As it's the same on prod I'm gonna open a follow up PR that will address all these broken indentations. Thanks for highlighting it @sympatheticmoose

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Pull request merged**